### PR TITLE
feat: bump styled-components lib version examples

### PR DESCRIPTION
**Pull request by omariosouto**
Hi folks I know many people that use that example and it could be using the new version of React to be according with the NextJS v11 

## Documentation / Examples

- [ ] Make sure the linting passes
- [ ] Make sure that the bumped up versions are okay

### CODE REVIEW

**Review by leerob — COMMENTED** [review]
Why is `react-is` needed?

**Comment by omariosouto** [comment]
@leerob I just follow the FAQ from the migration guide: https://styled-components.com/docs/faqs#what-do-i-need-to-do-to-migrate-to-v5

In a project that I've been working on it was necessary and already was present in other examples

File: examples/with-styled-components-rtl/package.json [hunk 1]
```diff
@@ -8,13 +8,13 @@
   },
   "dependencies": {
     "next": "latest",
-    "react": "16.12.0",
-    "react-dom": "16.12.0",
-    "styled-components": "5.0.0",
-    "stylis-plugin-rtl": "1.0.0"
```
Comment (leerob) [inline examples/with-styled-components-rtl/package.json:14]: Should this have been deleted?

File: examples/with-styled-components-rtl/package.json [hunk 2]
```diff
@@ -8,13 +8,13 @@
...
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-is": "^17.0.2",
+    "styled-components": "^5.2.3"
```
Comment (omariosouto) [inline examples/with-styled-components-rtl/package.json:14]:
```suggestion
    "styled-components": "^5.2.3",
    "stylis-plugin-rtl": "2.0.2"
```

File: examples/with-styled-components-rtl/package.json [hunk 3]
```diff
@@ -8,13 +8,14 @@
...
     "next": "latest",
-    "react": "16.12.0",
-    "react-dom": "16.12.0",
-    "styled-components": "5.0.0",
-    "stylis-plugin-rtl": "1.0.0"
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
+    "react-is": "^17.0.2",
+    "styled-components": "^5.2.3",
+    "stylis-plugin-rtl": "2.0.2"
```
Comment (omariosouto) [inline examples/with-styled-components-rtl/package.json:15]:
```suggestion
    "stylis-plugin-rtl": "1.0.0"
```

File: examples/with-styled-components-rtl/package.json (hunk 1 above)
Comment (omariosouto) [reply examples/with-styled-components-rtl/package.json:14]: Sorry, I forgot the last commit, now it's correct

**Review by leerob — APPROVED** [review]

**Review by ijjk — APPROVED** [review]

**Pull request merged**